### DISCUSSION TITLE
Using the Polyhedron Demo from code

### DISCUSSION THREAD

**Discussion by maximecharriere** [CGAL/cgal#6334] [Q&A] [answered]
For development, I need to be able to display several 3D elements on the same scene. 
For the moment I generate .ply files that I then open in the Polyhedron demo or in MeshLab. 

I would like to be able to display my meshes directly from the code. There is the `CGAL::draw()` function, but it only works to display 1 mesh. 

The Polyhedron demo already has everything I need internally, do you think there is a way to use it?
For example, would it be possible to launch the `Polyhedron_3.exe` program from the code and provide it with the meshes to display?
Or is there some kind of SDK in the demo to easily create a scene, add meshes and display it?

**Comment by lrineau** [comment, marked as answer]
Hi Maxime, I do not know exactly what you would like to do. Do you just want to launch the Polyhedron demo using your C++ code (and in that case [`std::system`](https://en.cppreference.com/w/cpp/utility/program/system) or Qt [`QProcess`](https://doc.qt.io/qt-5/qprocess.html) class might be answers). If you give several files to the demo as command line arguments, they will all be opened in the same scene at startup.

Or maybe you want to embed the Polyhedron demo in your own application. That is probably doable with slight modifications of CGAL, but that will be more work.

**Reply by maximecharriere** [reply]
Okay thank you for this answer! I didn't think the demo could be run from the command line with files as input. I will use this with [`std::system`](https://en.cppreference.com/w/cpp/utility/program/system).  
I'll see later if I have time to embed the Polyhedron demo in your own application.

**Comment by FuzhangHe** [comment]
Hi Irineau, 

A question about qt: 
I want to develop a visualization program like the CGAL polyhedron demo. 
I want to implement my own functions in the program. 

But there are so many source files in that demo programm, that I do not really understand them all. 

Can you explain how I should start?
